packages/legend-studio/src/stores/sidebar-state/LocalChangesState.ts

```diff
@@ -293,7 +467,42 @@ export class LocalChangesState {
     if (!localChanges.length) {
       return;
     }
-    this.isSyncingWithWorkspace = true;
+    yield flowResult(
+      this.sdlcState.fetchRemoteWorkspaceRevision(
+        this.sdlcState.activeProject.projectId,
+        this.sdlcState.activeWorkspace,
+      ),
+    );
+    if (this.sdlcState.isWorkspaceOutOfSync) {
+      this.editorStore.setActionAltertInfo({
+        message: 'Local workspace out-of-sync',
+        prompt: 'Please pull remote changes before pushing your local changes.',
+        type: ActionAlertType.CAUTION,
+        onEnter: (): void => this.editorStore.setBlockGlobalHotkeys(true),
+        onClose: (): void => this.editorStore.setBlockGlobalHotkeys(false),
+        actions: [
+          {
+            label: 'Pull Remote Changes',
```

Comment:
You're being a bit inconsistent with your letter casing here, let's do `Sentence case` -> `Pull remote changes`